browser/src/control/VRuler.js

```diff
@@ -0,0 +1,396 @@
+/* -*- js-indent-level: 8; fill-column: 100 -*- */
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+/*
+ * Ruler Handler
+ */
+
+/* global L _ */
+L.Control.VRuler = L.Control.extend({
```

Comment:
In the new code please avoid using Leaflet's code (L.Control / Class etc...).
We should use native `class` and also TypeScript.
Then we will be able to convert horizontal ruler and make some base class for both to share code.